### PR TITLE
Support more color escape codes

### LINKED ISSUE #743: unhandled osc_dispatch caused by base16 color-scheme
The base16-ocean.dark color scheme causes the following message to be printed by alacritty.

    [unhandled osc_dispatch]: [['1','2',],['r','g','b',':','c','0','/','c','5','/','c','e',],] at line 801

### PR DESCRIPTION
**Pull request by siiptuo**
This should fix #743. However some codes like those related to highlight color
are not implemented because they are not (yet) supported by the application.
Also intrestingly, there doesn't seem to be an escape code for changing cursor
foreground color, only the background color.

### CODE REVIEW

File: src/ansi.rs [hunk 1]
```diff
@@ -727,37 +744,20 @@ impl<'a, H, W> vte::Perform for Performer<'a, H, W>
...
-                    let mut index: u8 = 0;
-                    for c in raw {
-                        let c = *c as char;
-                        if let Some(digit) = c.to_digit(10) {
-                            index *= 10;
-                            index += digit as u8;
+                for chunk in params[1..].chunks(2) {
+                    if let Some(index) = parse_number(chunk[0]) {
+                        if let Some(color) = parse_rgb_color(chunk[1]) {
+                            self.handler.set_color(index as usize, color);
```
Comment (jwilm) [inline src/ansi.rs]: We should really have some sort of range check on index before passing to `set_color`. Is this OSC supposed to support indices outside of the u8 range?

File: src/ansi.rs [hunk 2]
```diff
@@ -78,6 +78,23 @@ fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
     }
 }
 
+fn parse_number(input: &[u8]) -> Option<u8> {
+    if input.is_empty() {
+        return None;
+    }
+    let mut index: u8 = 0;
```
Comment (jwilm) [inline src/ansi.rs]: Might want to rename index to `num` or something similar since this is a general number parsing method.

**Comment by siiptuo** [comment]
I've added overflow check. Using `u8` should be fine because 256-color table is used.

**Comment by jwilm** [comment]
LGTM; thanks @siiptuo!

**Pull request merged**